### PR TITLE
add systemd notify support

### PR DESCRIPTION
**Pull request by rubensfig**

### CODE REVIEW

File: src/netlink/cnetlink.cc [hunk 1]
```diff
@@ -388,12 +389,14 @@ void cnetlink::handle_wakeup(rofl::cthread &thread) {
   case NL_STATE_INIT:
     init_subsystems();
     state = NL_STATE_RUNNING;
+    sd_notify(0, "READY=1");
```
Comment (DFritzsche) [inline src/netlink/cnetlink.cc:392]: notifying here doesn't mean baseboxd has exposed all switch portsyet , does it?

File: pkg/systemd/baseboxd.service.in [hunk 2]
```diff
@@ -3,10 +3,11 @@ Description=baseboxd
 After=network.target
 
 [Service]
-Type=simple
+Type=notify
 ExecStart=@bindir@/baseboxd
 EnvironmentFile=-@sysconfdir@/sysconfig/baseboxd
 Restart=always
+NotifyAcess=main
```
Comment (jklare) [inline pkg/systemd/baseboxd.service.in:10]:
```suggestion
NotifyAccess=main
```

**Comment by DFritzsche** [comment]
one more thing, we should also notify not-ready or something when the openflow connection has dropped and/or baseboxd has stopped working

**Comment by rubensfig** [comment]
" there is little value in signaling non-readiness, the only value daemons should send is "READY=1"."

From: https://manpages.debian.org/jessie/libsystemd-dev/sd_notify.3.en.html

**Comment by DFritzsche** [comment]
How are we going to handle a termination of the OF connection? Losing the OF connection will delete all ports, preventing configuring these and hence all scripts/tools that work on these ports will fail.

**Comment by hilmarm** [comment]
lgtm, passed testing

**Pull request merged**